packaging/os/pulp_repo.py

```diff
@@ -0,0 +1,568 @@
+        importer_ssl_client_cert=dict(),
+        importer_ssl_client_key=dict(),
+        serve_http=dict(default=False, type='bool'),
+        serve_https=dict(default=True, type='bool'),
+        publish_distributor=dict())
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True)
+
+    repo = module.params['name']
+    state = module.params['state']
+    pulp_host = module.params['pulp_host']
+    feed_url = module.params['feed_url']
+    repo_type = module.params['repo_type']
+    relative_url = module.params['relative_url']
+    add_export_distributor = module.params['add_export_distributor']
+    proxy_host = module.params['proxy_host']
+    proxy_port = module.params['proxy_port']
+    importer_ssl_ca_cert = module.params['importer_ssl_ca_cert']
+    importer_ssl_client_cert = module.params['importer_ssl_client_cert']
+    importer_ssl_client_key = module.params['importer_ssl_client_key']
+    serve_http = module.params['serve_http']
+    serve_https = module.params['serve_https']
+    publish_distributor = module.params['publish_distributor']
+
+    if (state == 'present') and (not relative_url):
+        module.fail_json(msg="When state is present, relative_url is required.")
+
+    server = pulp_server(module, pulp_host, repo_type)
+    server.set_repo_list()
+    repo_exists = server.check_repo_exists(repo)
+
+    changed = False
+
+    if state == 'absent' and repo_exists:
+        if not module.check_mode:
+            server.delete_repo(repo)
+        changed = True
+
+    if state == 'sync':
+        if not repo_exists:
+            module.fail_json(msg="Repository was not found. The repository can not be synced.")
+
+        if not module.check_mode:
+            server.sync_repo(repo)
+
+        changed = True
+
+    if state == 'publish':
+        if not repo_exists:
+            module.fail_json(msg="Repository was not found. The repository can not be published.")
+
+        if not module.check_mode:
+            server.publish_repo(repo, publish_distributor)
+
+        changed = True
+
+    if state == 'present':
+        if not repo_exists:
+            if not module.check_mode:
+                server.create_repo(
+                    repo_id=repo,
+                    relative_url=relative_url,
+                    feed_url=feed_url,
+                    serve_http=serve_http,
+                    serve_https=serve_https,
+                    proxy_host=proxy_host,
+                    proxy_port=proxy_port,
+                    ssl_ca_cert=importer_ssl_ca_cert,
+                    ssl_client_cert=importer_ssl_client_cert,
+                    ssl_client_key=importer_ssl_client_key,
+                    add_export_distributor=add_export_distributor)
+
+            changed = True
+
+        else:
+            # Check to make sure all the settings are correct
+            # The importer config gets overwritten on set and not updated, so we set the whole config at the same time.
```

Comment:
Please could you wrap this like? It would be really great if you could keep all lines in the document under 80 characters  to be fully PEP8 compliant.